kolibri/plugins/learn/assets/src/views/LearnImmersiveLayout.vue

```diff
@@ -219,6 +267,105 @@
       });
     },
     methods: {
+      getSidebarInfo() {
+        if (this.lessonId) {
+          this.fetchLessonSiblings();
+        } else {
+          if (this.content.id) {
+            this.fetchSiblings();
+          }
+        }
+      },
+      /** Returns a promise which will get the progress for a set of nodes and map it back onto
+       *  the nodes, which are then set to this.viewResourcesContents.
+       *  @modifies this.viewResourcesContents - Sets it to the value of nodes, mapped with
+       *  progress, if found
+       *  @returns {Promise} - Just returns a meaningless Promise in case you want to chain
+       *  synchronously
+       */
+      progressPromise(nodes) {
```

Comment:
It took me a few minutes to realize that this method is also setting `this.viewResourcesContents` to allow the nodes to be displayed - I see now that it's in the JSDoc comment, but a more descriptive name would help too.